libs/async_mpi/include/hpx/async_mpi/mpi_future.hpp

```diff
@@ -165,14 +181,14 @@ namespace hpx { namespace mpi { namespace experimental {
     HPX_EXPORT void poll();
 
     // -----------------------------------------------------------------
-    // This is not completely safe as it will return when the request list is
+    // This is not completely safe as it will return when the request vector is
     // empty, but cannot guarantee that other communications are not about
     // to be launched in outstanding continuations etc.
     inline void wait()
     {
         hpx::util::yield_while([]() {
             std::unique_lock<detail::mutex_type> lk(
-                detail::get_list_mtx(), std::try_to_lock);
+                detail::get_vector_mtx(), std::try_to_lock);
```

Comment:
`get_requests_mtx`?

Reply:
It's not supposed to be exposed to the user and I renamed it really because the request-vector used to be a list, but then I switched to vector. There is a (lock free) request-queue as well and so I prefer `get_vector_mtx` to 'get_requests_mtx` as it reminds me exactly which object we are getting the lock for.

Reply:
mind you, `get_request_vector_mtx` would work...

Reply:
Yeah, no problem. I wasn't thinking about users with this change, just ourselves. Feel free to keep the name if you feel like it. I just thought it would be appropriate to have `requests` in the name since that's what it protects.